sjb/config/test_cases/ci-kubernetes-aws-actuator.yml

```diff
@@ -0,0 +1,236 @@
+---
+parent: 'common/test_cases/minimal.yml'
+extensions:
+  actions:
+    - type: "host_script"
+      title: "Upload the default AWS credentiasl"
+      script: |-
+         ssh -F ${WORKSPACE}/.config/origin-ci-tool/inventory/.ssh_config openshiftdevel "mkdir ~/.aws"
+         scp -F ${WORKSPACE}/.config/origin-ci-tool/inventory/.ssh_config ~/.aws/credentials openshiftdevel:~/.aws
+    - type: "script"
+      title: "Install minikube"
+      script: |-
+        # Install minikube
+        curl -Lo minikube https://storage.googleapis.com/minikube/releases/v0.28.0/minikube-linux-amd64
+        chmod +x minikube
+        sudo mv minikube /usr/bin/
+        # Install kubectl
+        curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.10.0/bin/linux/amd64/kubectl
+        chmod +x kubectl
+        sudo mv kubectl /usr/bin/
+    - type: "script"
+      title: "Deploy kubernetes"
+      script: |-
+        sudo setenforce 0
+        # https://github.com/kubernetes/minikube/blob/master/docs/configuring_kubernetes.md
+        sudo minikube start --vm-driver=none --extra-config=kubelet.cgroup-driver=systemd
+    - type: "script"
+      title: "Build the machine controller"
+      script: |-
+        export GOPATH=/data
+        cd $GOPATH/src/github.com/openshift/cluster-api-provider-aws
+        go get -u github.com/openshift/imagebuilder/cmd/imagebuilder
+        sudo mv /data/bin/imagebuilder /usr/bin
+        sudo make -C cmd/machine-controller
+    - type: "script"
+      title: "Deploy manifests"
+      script: |-
+        export GOPATH=/data
+        cd $GOPATH/src/github.com/openshift/cluster-api-provider-aws/examples
+        sudo pip install awscli
+
+        set +x
+        cat << HEREDOC > addons.yaml
+        ---
+        apiVersion: v1
+        kind: Namespace
+        metadata:
+          name: test
+        ---
+        apiVersion: v1
+        kind: Secret
+        metadata:
+          name: aws-credentials-secret
+          namespace: test
+        type: Opaque
+        data:
+          awsAccessKeyId: $(echo -n $(aws configure get aws_access_key_id) | base64)
```

Comment:
This two lines has the verbose mode disabled so the aws credentials are not printed into the logs.

Reply:
is this happening on an ec2 machine? if so we could also apply a know iam role and let the aws client omitting the credentials

Reply:
I did not check this use case. Though, in future we may need more permissions which we might not want to share with other ec2 instances (since all ec2 instance are provisioned by the oct). So I still think it's better to use the explicit aws credentials. Later, we can switch to a different AWS account.